geonode/base/api/tests.py

```diff
@@ -1998,6 +1998,56 @@ def test_set_resource_thumbnail(self):
                 )
                 self.assertEqual(response.status_code, 200)
 
+    @patch("geonode.base.api.views.remove_thumb")
+    def test_delete_thumbnail(self, mock_remove_thumb):
+
+        resource = Dataset.objects.first()
+
+        # Set a thumbnail url and path
+        resource.thumbnail_url = "http://example.com/thumb/test.png"
+        resource.thumbnail_path = "thumb/test.png"
+        resource.save()
+
+        url = reverse("base-resources-delete-thumbnail", kwargs={"resource_id": resource.id})
+
+        # Anonymous user
+        response = self.client.post(url)
+        self.assertEqual(response.status_code, 403)
+
+        # Authenticated user (admin)
+        self.assertTrue(self.client.login(username="admin", password="admin"))
+
+        # Valid thumbnail removal
+        response = self.client.post(url)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data["message"], "Thumbnail deleted successfully.")
+        resource.refresh_from_db()
+        self.assertIsNone(resource.thumbnail_url)
+        self.assertIsNone(resource.thumbnail_path)
+        mock_remove_thumb.assert_called_once_with("test.png")
+
+        # Thumbnail already deleted
+        response = self.client.post(url)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data["message"], "The thumbnail URL field is already empty.")
+        self.assertFalse(response.data["success"])
+
+        # Invalid image format
+        resource.thumbnail_url = "http://example.com/media/thumb/test.txt"
+        resource.thumbnail_path = "thumb/test.txt"
+        resource.save()
+
+        response = self.client.post(url)
+        self.assertEqual(response.status_code, 400)
+        self.assertIn("not a valid image", response.data)
+
+        # Resource does not exist
+        invalid_url = reverse("base-resources-delete-thumbnail", kwargs={"resource_id": 9999})
+        response = self.client.post(invalid_url)
+        self.assertEqual(response.status_code, 404)
```

Comment:
Hi @Gpetrak i guess it could be worth to add a test where a user, tries to delete the thumbnail of a resource which does not belong to it.
like user `bobby` tries to delete the thumbnail of a resource owner by user `norm`. I suppose the user should not have the right to do it